### DISCUSSION TITLE
Canvas + gesture detector does not work in Row parent control

### DISCUSSION THREAD

**Discussion by BCheong1990** [flet-dev/flet#4503] [Q&A] [answered]
Hi, based on the example code, I am trying to canvas + gesture detector to work in a Row parent control but the gestures don't pick up.. Here are the codes. Can someone please shed some light on this matter?

import flet as ft
import flet.canvas as cv

class State:
    x: float
    y: float

state = State()

def main(page: ft.Page):
    page.title = "Flet Brush"

    def pan_start(e: ft.DragStartEvent):
        state.x = e.local_x
        state.y = e.local_y

    def pan_update(e: ft.DragUpdateEvent):
        cp.shapes.append(
            cv.Line(
                state.x, state.y, e.local_x, e.local_y, paint=ft.Paint(stroke_width=3)
            )
        )
        cp.update()
        state.x = e.local_x
        state.y = e.local_y

    cp = cv.Canvas(
        [
            cv.Fill(
                ft.Paint(
                    gradient=ft.PaintLinearGradient(
                        (0, 0), (600, 600), colors=[ft.Colors.CYAN_50, ft.Colors.GREY]
                    )
                )
            ),
        ],
        content=ft.GestureDetector(
            on_pan_start=pan_start,
            on_pan_update=pan_update,
            drag_interval=10,
        ),
        expand=False,
    )

    page.add(
        ft.Row(
            [
                ft.Container(
                    cp,
                    # border_radius=5,
                    # width=float("inf"),
                    # expand=True,
                )
            ]
        )
    )

ft.app(main)

**Comment by ndonkoHenri** [comment, marked as answer]
The code works on my end, I am able to draw on the canvas.

If it still doesnt work on your end, try constraining your controls, by:
- providing width/height props 
- or adding `expand=True` for it to fill the available space.

```python
import flet as ft
import flet.canvas as cv


class State:
    x: float
    y: float


state = State()


def main(page: ft.Page):
    page.title = "Flet Brush"

    def pan_start(e: ft.DragStartEvent):
        state.x = e.local_x
        state.y = e.local_y

    def pan_update(e: ft.DragUpdateEvent):
        cp.shapes.append(
            cv.Line(
                state.x, state.y, e.local_x, e.local_y, paint=ft.Paint(stroke_width=3)
            )
        )
        cp.update()
        state.x = e.local_x
        state.y = e.local_y

    cp = cv.Canvas(
        [
            cv.Fill(
                ft.Paint(
                    gradient=ft.PaintLinearGradient(
                        (0, 0), (600, 600), colors=[ft.Colors.CYAN_50, ft.Colors.GREY]
                    )
                )
            ),
        ],
        content=ft.GestureDetector(
            on_pan_start=pan_start,
            on_pan_update=pan_update,
            drag_interval=10,
            expand=True,
        ),
        expand=True,
    )

    page.add(
        ft.Row(
            [
                ft.Container(
                    content=cp,
                    expand=True,
                    
                )
            ],
            expand=True,
        )
    )


ft.app(main)
```

**Reply by BCheong1990** [reply]
Oh yes, it does work in your code. Let me study it in detail. Thanks for your help.
